src/variables.jl

```diff
@@ -5,6 +5,56 @@
 
 abstract type AbstractVariable end
 
+# Any fields can usually be either a number or an expression
+mutable struct VariableInfoExpr
+    haslb::Bool
+    lowerbound::Any
+    hasub::Bool
+    upperbound::Any
+    hasfix::Bool
+    fixedvalue::Any
+    hasstart::Bool
+    start::Any
+    binary::Any
+    integer::Any
+end
+
+function setlowerbound(info::VariableInfoExpr, lower, _error::Function)
```

Comment:
`setlowerbound_or_error` might be a clearer name. setters don't usually fail like this.